hermeto/core/package_managers/oci/models.py

```
@@ -0,0 +1,174 @@
+    def _validate_images_non_empty(cls, value: list[OciImage]) -> list[OciImage]:
+        if not value:
+            raise ValueError("At least one image must be declared in the lockfile")
+        return value
+
+    @model_validator(mode="after")
+    def _validate_no_duplicate_digests(self) -> "OciLockfile":
+        seen: set[str] = set()
+        for image in self.images:
+            key = f"{image.repository}@{image.digest}"
+            if key in seen:
+                raise ValueError(f"Duplicate image entry: {key}")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)